src/Workspaces/Core/Portable/FindSymbols/FindReferences/FindReferencesSearchEngine.cs

```diff
@@ -71,38 +76,97 @@ public Task FindReferencesAsync(ISymbol symbol, CancellationToken cancellationTo
     public async Task FindReferencesAsync(
         ImmutableArray<ISymbol> symbols, CancellationToken cancellationToken)
     {
-        var unifiedSymbols = new MetadataUnifyingSymbolHashSet();
-        unifiedSymbols.AddRange(symbols);
+        var channel = Channel.CreateUnbounded<Reference>(s_channelOptions);
```

Comment:
same channel approach, allowing hte finding to happel in parallel, pushing to the channel.  And a single reading task pulling off chunks and reporting it to listeners.